scripts/ligandtocore/QD.py

```diff
@@ -0,0 +1,203 @@
+                                             for name in dir_name_list]
+
+    # Imports the cores and ligands
+    core_list = qd_scripts.read_mol(core_folder, input_cores, column, row, smiles_extension)
+    ligand_list = qd_scripts.read_mol(ligand_folder, input_ligands, column, row, smiles_extension)
+
+    # Return the indices of the core dummy atoms
+    core_indices = [prep_core(core, core_folder, dummy, core_opt) for core in core_list]
+
+    # Open the ligand database and check if the specified ligand(s) is already present
+    if use_database:
+        database = qd_scripts.read_database(ligand_folder, database_name)
+    else:
+        database = [[], [], [], [], []]
+
+    ligand_list = [prep_ligand(ligand, ligand_folder, database, ligand_opt, split) for
+                   ligand in ligand_list]
+
+    # Formating of ligand_list
+    ligand_list, ligand_indices, database_entries = zip(*ligand_list)
+    ligand_indices = list(itertools.chain(*ligand_indices))
+    ligand_list = itertools.chain(*ligand_list)
+
+    # Write new entries to the ligand database
+    if use_database:
+        qd_scripts.write_database(database_entries, ligand_folder, database)
+
+    # Combine the core with the ligands, yielding qd
+    qd_list = [prep_qd(core, ligand, core_indices[i], ligand_indices[j], qd_folder) for i, core in
+               enumerate(core_list) for j, ligand in enumerate(ligand_list)]
+
+    # Formating of qd_list
+    qd_list, pdb_name_list, qd_indices = zip(*qd_list)
+
+    # Check if the ADF environment variables are set and optimize the qd with the core frozen
+    if qd_opt:
+        sys_var = ['ADFBIN', 'ADFHOME', 'ADFRESOURCES', 'SCMLICENSE']
+        sys_var_exists = [item in os.environ for item in sys_var]
+        for i, item in enumerate(sys_var_exists):
+            if not item:
+                print('WARNING: The environment variable ' + sys_var[i] + ' has not been set')
+        if False in sys_var_exists:
+            raise MoleculeError('One or more ADF environment variables have not been set,' +
+                                ' aborting geometry optimization.')
+        for i, qd in enumerate(qd_list):
+            qd_scripts.prep_ams_job(qd, pdb_name_list[i], qd_folder, qd_indices[i], maxiter)
+
+    # The End
+    time_end = time.time()
+    print('\nTotal elapsed time:\t\t' + '%.4f' % (time_end - time_start) + ' sec')
+
+
+
+# Argument list
+path = r'/Users/basvanbeek/Documents/CdSe/Week_5'
+dir_name_list = ['core', 'ligand', 'QD']
+input_cores = 'Cd68Se55.xyz'
+input_ligands = ['CCCCCCCCC([O-])=O.CC[N+](CC)(CC)CC', 'OCCCCCCCCC']
+smiles_extension = '.txt'
+column = 0
+row = 0
+dummy = 'Cl'
+database_name = 'ligand_database.txt'
+use_database = True
+core_opt = False
+ligand_opt = True
+qd_opt = False
+maxiter = 10000
+split = True
+
+prep_prep_args = [path, dir_name_list, input_cores, input_ligands, smiles_extension, column, row,
```

Comment:
@BvB93 A list of arguments is error prone, a better way is to use a dictionary like:
```
prep_prep_args = {'path': 'some/path', ...}
````